darts/explainability/shap_explainer.py

```diff
@@ -0,0 +1,635 @@
+                        shap_[h][t].time_index,
+                        shap_[h][t].values,
+                        columns=shap_[h][t].feature_names,
+                    )
+                shap_values_dict[h] = tmp
+
+            shap_values_list.append(shap_values_dict)
+
+        if len(shap_values_list) == 1:
+            shap_values_list = shap_values_list[0]
+
+        return shap_values_list
+
+    def summary_plot(
+        self,
+        horizons: Optional[Sequence[int]] = None,
+        target_names: Optional[Sequence[str]] = None,
+        nb_samples: Optional[int] = None,
+        plot_type: Optional[str] = "dot",
+    ):
+        """
+        Display a shap plot summary per (target, horizon)).
+        We here reuse the initial background data as foreground (potentially sampled) to give a general importance
+        plot for each feature.
+        If no target names and/or no horizons are provided, we plot all summary plots.
+
+        Parameters
+        ----------
+        horizons
+            Optionally, a list of integer values representing which elements in the future
+            we want to explain, starting from the first timestamp prediction at 0.
+            For now we consider only models with output_chunk_length and it can't be bigger than output_chunk_length.
+        target_names
+            Optionally, A list of string naming the target names we want to plot.
+        nb_samples
+            Optionally, an integer value sampling the foreground series (based on the backgound),
+            for the sake of performance.
+        plot_type
+            Optionally, string value for the type of plot proposed by shap library. Currently,
+            the following are available: 'dot', 'bar', 'violin'.
+
+        """
+
+        horizons, target_names = self._check_horizons_and_targets(
+            horizons, target_names
+        )
+
+        if nb_samples:
+            foreground_X_sampled = shap.utils.sample(
+                self.explainers.background_X, nb_samples
+            )
+        else:
+            foreground_X_sampled = self.explainers.background_X
+
+        shaps_ = self.explainers.shap_explanations(
+            foreground_X_sampled, horizons, target_names
+        )
+
+        for t in target_names:
+            for h in horizons:
+                plt.title("Target: `{}` - Horizon: {}".format(t, "t+" + str(h)))
+                shap.summary_plot(
+                    shaps_[h][t], foreground_X_sampled, plot_type=plot_type
+                )
+
+    def force_plot_from_ts(
+        self,
+        foreground_series: TimeSeries,
+        foreground_past_covariates: Optional[TimeSeries] = None,
+        foreground_future_covariates: Optional[TimeSeries] = None,
+        horizons: Optional[Sequence[int]] = None,
+        target_names: Optional[Sequence[str]] = None,
+    ):
+        """
+        Display a shap force_plot per target and per horizon.
```

Comment:
It would be nice to put a couple of sentences explaining what the force plot shows (in this context) for people not familiar with Shap.